code/datums/progressbar.dm

```diff
@@ -59,9 +63,13 @@
 	if(!bars.len)
 		LAZYREMOVE(user.progressbars, bar.loc)
 
-	if (client)
-		client.images -= bar
-	qdel(bar)
+	animate(bar, alpha = 0, time = 5)
+	addtimer(CALLBACK(src, /datum/progressbar/proc/remove_from_client), 5)
```

Comment:
This needs `TIMER_CLIENT_TIME`

when you use `animate`, you need more precise timers.